pkg/plugins/manager/loader/angular/angulardetector/gcom.go

```diff
@@ -0,0 +1,204 @@
+		// Fail silently
+		p.log.Warn("Could not update dynamic detectors", "error", err)
+	}
+	return p.detectors
+}
+
+// fetch fetches the angular patterns from GCOM and returns them as gcomPatterns.
+// Call detectors() on the returned value to get the corresponding detectors.
+func (p *GCOMDetectorsProvider) fetch(ctx context.Context) (gcomPatterns, error) {
+	st := time.Now()
+
+	reqURL, err := url.JoinPath(p.baseURL, gcomAngularPatternsPath)
+	if err != nil {
+		return nil, fmt.Errorf("url joinpath: %w", err)
+	}
+
+	p.log.Debug("Fetching dynamic angular detection patterns", "url", reqURL)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("new request with context: %w", err)
+	}
+	resp, err := p.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("http do: %w", err)
+	}
+	defer func() {
+		if closeErr := resp.Body.Close(); closeErr != nil {
+			p.log.Error("response body close error", "error", err)
+		}
+	}()
+	var out gcomPatterns
+	if err := json.NewDecoder(resp.Body).Decode(&out); err != nil {
+		return nil, fmt.Errorf("json decode: %w", err)
+	}
+	p.log.Debug("Fetched dynamic angular detection patterns", "patterns", len(out), "duration", time.Since(st))
+	return out, nil
+}
+
+// gcomPatternType is a pattern type returned by the GCOM API.
+type gcomPatternType string
+
+const (
+	gcomPatternTypeContains gcomPatternType = "contains"
+	gcomPatternTypeRegex    gcomPatternType = "regex"
+)
+
+// gcomPattern is an Angular detection pattern returned by the GCOM API.
+type gcomPattern struct {
+	Name    string
+	Pattern string
+	Type    gcomPatternType
+}
+
+// errUnknownPatternType is returned when a pattern type is not known.
+var errUnknownPatternType = errors.New("unknown pattern type")
+
+// Detector converts a gcomPattern into a Detector, based on its Type.
+// If a pattern type is unknown, it returns an error wrapping errUnknownPatternType.
+func (p *gcomPattern) detector() (Detector, error) {
+	switch p.Type {
+	case gcomPatternTypeContains:
+		return &ContainsBytesDetector{Pattern: []byte(p.Pattern)}, nil
+	case gcomPatternTypeRegex:
+		re, err := regexp.Compile(p.Pattern)
+		if err != nil {
+			return nil, fmt.Errorf("%q regexp compile: %w", p.Pattern, err)
+		}
+		return &RegexDetector{Regex: re}, nil
+	}
+	return nil, fmt.Errorf("%q: %w", p.Type, errUnknownPatternType)
+}
+
+// gcomPatterns is a slice of gcomPattern s.
+type gcomPatterns []gcomPattern
+
+// detectors converts the slice of gcomPattern s into a slice of detectors, by calling Detector() on each gcomPattern.
+func (p gcomPatterns) detectors() ([]Detector, error) {
+	var finalErr error
+	detectors := make([]Detector, 0, len(p))
+	for _, pattern := range p {
+		d, err := pattern.detector()
+		if err != nil {
+			// Fail silently in case of an errUnknownPatternType.
+			// This allows us to introduce new pattern types without breaking old Grafana versions
```

Comment:
should at least debug the error?

Reply:
Yep good idea, however it's a bit hard to do now because we have no logger in there current. I can leave it for a folow-up PR maybe?

Reply:
Up to you because it's so small you may forget. It should be easy to add, just make a struct arround gcomPatterns and forward the logger from GCOMDetectorsProvider.